codenvy-api-factory/src/main/java/com/codenvy/api/factory/FactoryService.java

```diff
@@ -292,6 +295,56 @@ public Factory getFactory(@ApiParam(value = "Factory ID", required = true)
     }
 
     /**
+     * Removes factory information from storage by its id.
+     *
+     * @param id id of factory
+     * @param uriInfo
+     *         - url context
+     * @throws NotFoundException when factory with given id doesn't exist
+     */
+    @ApiOperation(value = "Removes Factory information by its ID",
+            notes = "Removes factory based on the Factory ID which is passed in a path parameter")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "OK"),
+            @ApiResponse(code = 403, message = "User not authorized to call this operation"),
+            @ApiResponse(code = 404, message = "Factory not found"),
+            @ApiResponse(code = 500, message = "Internal Server Error")})
+    @DELETE
+    @Path("/{id}")
+    public void removeFactory(@ApiParam(value = "Factory ID", required = true)
+                              @PathParam("id") String id,
+                              @Context UriInfo uriInfo) throws ForbiddenException, NotFoundException, ApiException {
+        Factory factory = factoryStore.getFactory(id);
+        if (factory == null) {
+            LOG.warn("Factory URL with id {} is not found.", id);
+            throw new NotFoundException("Factory with id " + id + " is not found.");
+        }
+
+        // Check user has the right to delete this factory (needs to have created it)
+        final User user = EnvironmentContext.getCurrent().getUser();
+        if (user == null) {
+            throw new ForbiddenException("User needs to be logged in to remove a factory");
+        }
+        String userId = user.getId();
+        if (userId == null || userId.isEmpty()) {
+            throw new ApiException("Invalid factory without author stored");
```

Comment:
Check for userid in EnvironmentContext not factory, but message says "Invalid factory ... "
In such cases throwing server exception instead of general api exception would be better
User in EnvironmentContext must contain id, so this check is not needed( but user object may be null)